core/src/main/scala/org/apache/spark/TaskContextImpl.scala

```diff
@@ -59,8 +59,8 @@ private[spark] class TaskContextImpl(
   /** List of callback functions to execute when the task fails. */
   @transient private val onFailureCallbacks = new ArrayBuffer[TaskFailureListener]
 
-  // Whether the corresponding task has been killed.
-  @volatile private var interrupted: Boolean = false
+  // If defined, the corresponding task has been killed for the contained reason.
+  @volatile private var maybeKillReason: Option[String] = None
```

Comment:
nit: Overloading `maybeKillReason` to indicate  `interrupted` status smells a bit; but might be ok for now.